catkin_ws/src/state_estimation/src/state_aggregator.py

```diff
@@ -29,8 +33,34 @@ def update_state(_):
             break
     for sensor in sensor_priorities["z"]:
         if sensor.isActive():
-            z = sensor.z
-            break
+            transition_ds_dvl = rospy.get_param("~transition_ds_dvl")
+            if transition_ds_dvl:
+                if sensor.sensor_name == "depth_sensor":
+                # reads z value from the sensor 
+                    ds_z = sensor.z
+                    # if switch has been done, the offset is ready, shift the ds value by the offset
+                    # the offset converges to 0 as long as ds works and doesn't switch to the dvl
+                    if dvl_z is not None: #if switch has occurred
+                        # when i increases, exp decreases and converges to 0
+                        i += 1
+                        ds_z -= math.exp(-i)*offset
+                    z = ds_z
+                    break
+                else: # dvl
+                    # re-calculate offset for each switch
+                    dvl_z = sensor.z
+                    if i != 0:
```

Comment:
I think `i` needs to be initialized to a non-zero value (if the depth sensor is active from the start the dvl should grab an offset even if the depth sensor has not transitioned yet)

Reply:
So should i just initialize i to (for example) 1, and then keep the same condition and the reset to 0?